src/v/archival/ntp_archiver_service.cc

```diff
@@ -506,8 +485,9 @@ ss::future<> ntp_archiver::upload_topic_manifest() {
           &_rtcnode);
         retry_chain_logger ctxlog(archival_log, fib);
         vlog(ctxlog.info, "Uploading topic manifest {}", _parent.ntp());
-        cloud_storage::topic_manifest tm(
-          convert_topic_configuration(topic_cfg, replication_factor), _rev);
+        auto cfg_copy = topic_cfg.get();
+        cfg_copy.replication_factor = replication_factor;
```

Comment:
wondering what's the default value of the replication_factor if we don't set it?

Reply:
it should in principle match the correct replication factor for a topic. but I don't have context as to why then the replication factor is recomputed from the number of voters, few lines above

```cpp
   auto replication_factor = cluster::replication_factor{
      _parent.raft()->config().current_config().voters.size()};
```